File: FrequenciaNumerica/FrequenciaNumerica.java
import java.util.Scanner;

public class FrequenciaNumerica{
    
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int vetor[] = new int[2000];
        int n;

        for(int i = 0; i < 2000; i ++)
            vetor[i] = -1;
        
        n = sc.nextInt();

        for(int i = 0; i < n; i++)
            vetor[sc.nextInt()]++;
            
        for(int i = 0; i < 2000; i++){
            if(vetor[i] != -1){
                System.out.printf("%d aparece %d vez(es)\n",i,vetor[i]+1);
            }
        }

        sc.close();
    }
}

File: FrequenciaNumerica/FrequenciaNumerica.c
#include <stdio.h>

int main(){
    int vetor[2000];
    int n, aux;
        
        //Preenche todas as posições do vetor com zero
        for(int i = 0; i < 2000; i ++)
            vetor[i] = 0;
        
        scanf("%d",&n); 

        //o valor informado é usado como indice do vetor e incrimenta-se o valor indicado pelo indice.
        for(int i = 0; i < n; i++){
            scanf("%d",&aux);
            vetor[aux]++;
        }

        //Imprime apenas os vetores que possuem valor diferente de zero.  
        for(int i = 0; i < 2000; i++){
            if(vetor[i] != 0){
                printf("%d aparece %d vez(es)\n",i,vetor[i]);
            }
        }
    
    return 0;
}